crates/starknet-devnet/src/cli.rs

```diff
@@ -237,19 +244,55 @@ impl Args {
                 url: self.fork_network.clone(),
                 block_number: self.fork_block,
             },
-            disable_account_impersonation: self.disable_account_impersonation,
         };
 
         let RequestResponseLogging { log_request, log_response } =
             RequestResponseLogging::from_rust_log_environment_variable();
 
+        // if restricted_methods are not specified, use default ones
+        let restricted_methods = self.restricted_methods.as_ref().map(|methods| {
+            if methods.is_empty() {
+                DEFAULT_RESTRICTED_JSON_RPC_METHODS
+                    .iter()
+                    .map(|s| s.to_string())
+                    .collect::<Vec<String>>()
+            } else {
+                // remove leading slashes
+                methods
+                    .iter()
+                    .map(|s| s.trim_start_matches('/').to_string())
+                    .collect::<Vec<String>>()
+            }
+        });
+        // validate restricted methods
+        if let Some(methods) = restricted_methods.as_ref() {
+            let json_rpc_methods = JsonRpcRequest::all_variants_serde_renames();
+            let all_methods: HashSet<_> = HashSet::from_iter(
+                json_rpc_methods.iter().chain(HTTP_API_ROUTES_WITHOUT_LEADING_SLASH.iter()),
+            );
+            let mut wrong_restricted_methods = vec![];
+            for method in methods {
+                if !all_methods.contains(method) {
+                    wrong_restricted_methods.push(method.clone());
+                }
+            }
+            if !wrong_restricted_methods.is_empty() {
+                anyhow::bail!(
+                    "Restricted methods contain JSON-RPC methods and/or HTTP routes that are not \
+                     supported by the server: {}",
+                    wrong_restricted_methods.join(" ")
```

Comment:
Why not comma-separate them? I.e. `.join(", ")`

Reply:
because they are space separated when specified. Keeping the same format

Reply:
makes sense